flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/UserDefinedFunctionUtils.scala

```diff
@@ -358,4 +365,123 @@ object UserDefinedFunctionUtils {
     InstantiationUtil
       .deserializeObject[UserDefinedFunction](byteData, Thread.currentThread.getContextClassLoader)
   }
+
+  /**
+    * Build a TableFunctionCall, a name and a sequence of params will determine a unique
+    * [[TableFunctionCall]]
+    *
+    * @param name function name
+    * @param implicitResultType If no result type returned, it will use this type.
+    * @param params The input expressions
+    * @return A unique [[TableFunctionCall]]
+    */
+  private[table] def buildTableFunctionCall(
+      name: String,
+      tableFunction: TableFunction[_],
+      implicitResultType: TypeInformation[_],
+      params: Expression*): TableFunctionCall = {
+    val arguments = transformLiteralExpressions(params: _*)
+    val classes = params.map { param =>
```

Comment:
This extracts the classes of the arguments, not the classes of the arguments of the `eval()` signature.